theano/tensor/nnet/abstract_conv.py

```diff
@@ -216,24 +217,50 @@ def conv2d_grad_wrt_inputs(output_grad,
 
     """
 
-    grad_input_op = AbstractConv2d_gradInputs(imshp=input_shape,
+    filters = as_tensor_variable(filters)
+    output_grad = as_tensor_variable(output_grad)
+
+    # checking the type of input_shape
+    for dim in [0, 1]:
+        assert isinstance(input_shape[dim], (theano.tensor.TensorConstant,
+                                             int, type(None)))
+    for dim in [2, 3]:
+        assert isinstance(input_shape[dim], (theano.tensor.TensorVariable,
+                                             theano.tensor.TensorConstant,
+                                             int))
+
+    # checking the type of filter_shape
+    for dim in [0, 1, 2, 3]:
+        assert isinstance(filter_shape[dim], (theano.tensor.TensorVariable,
```

Comment:
In my code in bilinear_upsampling code, for one of the dimensions of filter_dim, I'm using a tensor variable and it is working fine. I think tensor_variable can be used, but not sure.

Another comment, if the user just passes None (as the default value), I guess this line would fail since it's neither a list nor a tuple.